gitlab/v4/objects/projects.py

```diff
@@ -1152,13 +1159,9 @@ def import_github(
             print("Github import complete")
 
         """
-        data = {
-            "personal_access_token": personal_access_token,
-            "repo_id": repo_id,
-            "target_namespace": target_namespace,
-        }
-        if new_name:
-            data["new_name"] = new_name
+        attributes = copy.deepcopy(locals())
+        data = utils.remove_none_from_dict(attributes, ignore_keys=("self", "kwargs"))
```

Comment:
Not sure about this but I was getting a bit tired of the custom code we need when we add arguments. Seems cleaner this way while preserving the method signature with typing.

Reply:
Yeah. I'm not sure about this either. I think it makes the code harder to understand. It is cool 😎.  But I think the other way is easier to understand.

So basically I don't think we should do it this way.

Reply:
What about something like this?

```suggestion
        data = {
            k: v
            for k, v in [
                ("personal_access_token", personal_access_token),
                ("repo_id", repo_id),
                ("target_namespace", target_namespace),
                ("new_name", new_name),
                ("github_hostname", github_hostname),
                ("optional_stages", optional_stages),
            ]
            if v is not None
        }
```

Reply:
This type of duplication is what I was trying to avoid.. Ok, I'll take another look at what options we might have for handling this in a generic way :) Since we also have other custom methods where the explicit arguments lag behind GitLab's implementation.

Reply:
For now I reverted the fancy stuff but reused the helper function. :P